src/libstd/sys/unix/stack_overflow.rs

```diff
@@ -169,13 +169,27 @@ mod imp {
     }
 
     pub unsafe fn make_handler() -> Handler {
-        let stack = get_stack();
-        sigaltstack(&stack, ptr::null_mut());
-        Handler { _data: stack.ss_sp as *mut libc::c_void }
+        let mut stack = mem::zeroed();
+        sigaltstack(ptr::null(), &mut stack);
+        if stack.ss_flags & SS_DISABLE != 0 {
```

Comment:
Could you add a comment explaining what this check is doing? I think it's just checking to see if one's already set, right?